docs/manual/codespaces.md

```diff
@@ -0,0 +1,273 @@
+
+/// tab | Button
+<div align=center markdown>
+<a href="https://codespaces.new/srl-labs/srlinux-vlan-handling-lab?quickstart=1">
+<img src="https://gitlab.com/rdodin/pics/-/wikis/uploads/d78a6f9f6869b3ac3c286928dd52fa08/run_in_codespaces-v1.svg?sanitize=true" style="width:50%"/></a>
+
+**[Run](https://codespaces.new/srl-labs/srlinux-vlan-handling-lab?quickstart=1) this lab in GitHub Codespaces for free**.  
+[Learn more](https://containerlab.dev/manual/codespaces) about Containerlab for Codespaces.  
+<small>Machine type: 2 vCPU · 8 GB RAM</small>
+</div>
+///
+/// tab | Button code
+**Do not forget to change the lab repo URL and machine type in the code below!**
+
+```html
+---
+<div align=center markdown>
+<a href="https://codespaces.new/srl-labs/srlinux-vlan-handling-lab?quickstart=1">
+<img src="https://gitlab.com/rdodin/pics/-/wikis/uploads/d78a6f9f6869b3ac3c286928dd52fa08/run_in_codespaces-v1.svg?sanitize=true" style="width:50%"/></a>
+
+**[Run](https://codespaces.new/srl-labs/srlinux-vlan-handling-lab?quickstart=1) this lab in GitHub Codespaces for free**.  
+[Learn more](https://containerlab.dev/manual/codespaces) about Containerlab for Codespaces.  
+<small>Machine type: 2 vCPU · 8 GB RAM</small>
+</div>
+
+---
+```
+
+///
+
+Check out [srl-labs/srl-streaming-telemetry](https://github.com/srl-labs/srl-telemetry-lab?tab=readme-ov-file#nokia-sr-linux-streaming-telemetry-lab) README where this button is used to start the lab in Codespaces.
+
+And of course, you can always launch the Codespace using the GitHub UI by clicking on the "Code" button.
+
+![start](https://gitlab.com/rdodin/pics/-/wikis/uploads/34a9e417d61c02cf9a4283cc1c2cf8cb/image.png)
+
+1. Click on the "Code" button in the GitHub UI.
+2. Start the Codespace on the `main` branch using the devcontainer settings defined in the `.devcontainer/devcontainer.json` file.
+3. Or open up an advanced menu
+4. And configure the Codespace settings manually.
+
+## Dev Container
+
+The key pillar behind Codespaces is the Containerlab' Dev Container image that defines the environment in which the lab will run. The Dev Container image is a Docker image that contains all the necessary tools and dependencies to run Containerlab and other tools that you might need in the lab.
+
+The following two files define Containerlab's Dev Container image:
+
+1. [devcontainer.json](https://github.com/srl-labs/containerlab/blob/main/.devcontainer/devcontainer.json) - the Dev Container configuration file that defines how the environment is built, configured and launched.
+2. [Dockerfile](https://github.com/srl-labs/containerlab/blob/main/.devcontainer/Dockerfile) - the Dockerfile that the Dev Container is built from.
+
+The resulting Dev Container image contains the following tools and dependencies:
+
+- containerlab binary installed via the deb repository
+- docker in docker setup
+- [gNMIc](https://gnmic.openconfig.net) and [gNOIc](https://gnoic.kmrd.dev) tools
+- Go SDK
+- Python 3 with pyenv
```

Comment:
No pyenv please!! =) I think, this can hurt in future. Happy to be wrong though.

Reply:
what harm you see in having pyenv?
If one doesn't need it - they are free not to use it. The pyenv itself is a lightweight wrapper, it doesn't consume anything from the resources.

Reply:
Totally up to you. My point of view is that container must be complete and closed environment with all dependencies installed inside and fixed path to everything. venv/pyenv open... possibilities. =) You can have Python3.10 on a Python3.11 container, pointing to a different path and sometimes even sourced from the repo directly. I have seen quite a few weird things happening while helping to troubleshoot. Therefore my preference to have all dependancies as part of container build is very strong. However in ideal case we should still provide some options by building multiple containers with different tags, Python versions, etc. As that will be separate matrix PR somewhere in the future - we can remove Pyenv once there is an alternative.

Reply:
I agree, we need to look at the `-slim` images next where everything dev related is stripped completely.

For now the `<version>` tag will contain the dev tools such as pyenv